File: core/ui/src/types/employee.ts
export type EmployeeStatus = "ACTIVE" | "ON_LEAVE" | "INACTIVE" | "TERMINATED";

export type EmployeeQualificationStatus = "PENDING" | "QUALIFIED" | "TRAINING_DUE" | "SUSPENDED";

export type Employee = {
  id: string;
  employeeCode: string;
  fullName: string;
  email: string | null;
  phone: string | null;
  department: string;
  site: string | null;
  jobTitle: string;
  managerEmployeeId: string | null;
  managerEmployeeCode: string | null;
  managerName: string | null;
  employmentStatus: EmployeeStatus;
  qualificationStatus: EmployeeQualificationStatus;
  joinedOn: string | null;
  lastTrainingDate: string | null;
  nextTrainingDue: string | null;
  remarks: string | null;
  isActive: boolean;
  createdBy: string;
  createdAt: string;
  updatedBy: string | null;
  updatedAt: string | null;
};

export type EmployeeRequest = {
  employeeCode: string;
  fullName: string;
  email?: string | null;
  phone?: string | null;
  department: string;
  site?: string | null;
  jobTitle: string;
  managerEmployeeId?: string | null;
  employmentStatus?: EmployeeStatus;
  qualificationStatus?: EmployeeQualificationStatus;
  joinedOn?: string | null;
  lastTrainingDate?: string | null;
  nextTrainingDue?: string | null;
  remarks?: string | null;
  createdBy?: string;
  updatedBy?: string;
};

File: core/src/main/java/com/batchsphere/core/hrms/employee/entity/Employee.java
package com.batchsphere.core.hrms.employee.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "employee")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Employee {

    @Id
    @Column(nullable = false, updatable = false)
    private UUID id;

    @Column(name = "employee_code", nullable = false, unique = true, length = 50)
    private String employeeCode;

    @Column(name = "full_name", nullable = false, length = 150)
    private String fullName;

    @Column(unique = true, length = 150)
    private String email;

    @Column(length = 40)
    private String phone;

    @Column(nullable = false, length = 100)
    private String department;

    @Column(length = 120)
    private String site;

    @Column(name = "job_title", nullable = false, length = 120)
    private String jobTitle;

    @Column(name = "manager_employee_id")
    private UUID managerEmployeeId;

    @Enumerated(EnumType.STRING)
    @Column(name = "employment_status", nullable = false, length = 40)
    private EmployeeStatus employmentStatus;

    @Enumerated(EnumType.STRING)
    @Column(name = "qualification_status", nullable = false, length = 40)
    private EmployeeQualificationStatus qualificationStatus;

    @Column(name = "joined_on")
    private LocalDate joinedOn;

    @Column(name = "last_training_date")
    private LocalDate lastTrainingDate;

    @Column(name = "next_training_due")
    private LocalDate nextTrainingDue;

    @Column(length = 1000)
    private String remarks;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive;

    @Column(name = "created_by", nullable = false, length = 100)
    private String createdBy;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_by", length = 100)
    private String updatedBy;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
